### ISSUE TITLE
[LLVM 3.9.1] AssertionError: TypeKind.ELABORATED

### ISSUE THREAD

**Issue by dmikushin** [osandov/ctypesgen#1]
Hi @osandov ! With LLVM `3.6.2` the tests for your `ctypesgen` project are passing. But for LLVM version `3.9.1` the tests no longer pass with identical errors like this one:

```
 > [13/13] RUN ./run_tests.sh:                                                                                                                                
0.359 tests/anonymous.h                                                                                                                                       
0.409 Traceback (most recent call last):                                                                                                                      
0.409   File "./ctypesgen.py", line 400, in <module>                                                                                                          
0.409     main()                                                                                                                                              
0.409   File "./ctypesgen.py", line 79, in main                                                                                                               
0.409     walk(translation_unit.cursor)                                                                                                                       
0.409   File "./ctypesgen.py", line 203, in walk                                                                                                              
0.409     print_decl(top)                                                                                                                                     
0.409   File "./ctypesgen.py", line 276, in print_decl                                                                                                        
0.409     decl.print_definition()                                                                                                                             
0.409   File "./ctypesgen.py", line 137, in print_definition                                                                                                  
0.409     ctype = clang_type_to_ctype(field.type)                                                                                                             
0.409   File "./ctypesgen.py", line 396, in clang_type_to_ctype                                                                                               
0.409     assert False, type.kind                                                                                                                             
0.409 AssertionError: TypeKind.ELABORATED
```

Do you have an advise for me what to do with this `AssertionError: TypeKind.ELABORATED`?

**Comment by dmikushin** [comment]
The newest LLVM version which ctypegen runs fine with is `3.8.1`.

**Comment by dmikushin** [comment]
With the recent followup commits also landed into #2 , ctypesgen handles the `AssertionError: TypeKind.ELABORATED` properly.
